app/controllers/katello/concerns/api/v2/bulk_hosts_extensions.rb

```diff
@@ -3,37 +3,41 @@ module Concerns
     module Api::V2::BulkHostsExtensions
       extend ActiveSupport::Concern
 
+      def bulk_hosts_relation(permission, org)
```

Comment:
@parthaa after testing it, I don't think it makes much sense to throw error about the missing ids here. The reason being is that the scoped search or restrict parameters can alter the output and so ids which were included may ultimately be excluded. Opted to keep current behavior, but I did a few things to clean this concern up:

- fail at the start if no ids or search given
- framed a few of the `unless` into `if` for comprehension
- de-duped some code for the hosts relation
- used `merge` for the query when a scoped search is given + other improvement in that area

None of the tests broke, and I added another one for the .merge usage since we didn't have coverage in that case